interview-questions.md

```diff
@@ -7,44 +7,70 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 3. Name three rails generator commands. What is created by each?
 
-  Your answer:
+  Your answer: Three rails generator commands are rails generate controller, rails generate migration, rails generate model. Rails generate controller creates a file where we can create restful route methods and the logic we plan to associate with them. Rails generate migration lets us create an update to our schema. Rails generate model is what actually builds the schema.
 
   Researched answer:
+  In the rails generate model come, not only do we have to name our model, but state the columns. These columns will be paired with the expectation of what data type each of these categories accept as arguments. If we forget or want to change any, we can also generate and execute a migration later. Once this command is complete, it creates the schema of the database. Since the model is a sort of class, the name needs to be upper case. In class we learned that it needs to be singular, but with further research, I think it's because plurals are reserved for controller names.
+
+  As far as controller goes, I remember reading something about MVC being a compared to a restaurant, and the controller is what communicates what's in the model to the user in the view. Further still, the controller knows HOW to communicate with the model while the user in the view doesn't have that capability.
+
+
 
 
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
 action: "GET"    location: /students          
+The INDEX method would display all of the rows in the database.
 
 action: "POST"   location: /students       
+The CREATE method would be passed arguments and save a new object row to the student database.
 
 action: "GET"    location: /students/new
+The NEW method would create a new object row like the rows in our database, but won't finalize it unless it's saved.
 
 action: "GET"    location: /students/2  
+The SHOW method would display the student with the primary key of 2.
 
-action: "GET"    location: /students/2/edit    
+action: "GET"    location: /students/2/edit  
+The EDIT method would generate a form to submit changes to the student at primary key 2.
 
 action: "PATCH"  location: /students/2      
+The UPDATE method would be passed changes as arguments. The row with the primary key "2" is what the arguments would update.
 
 action: "DELETE" location: /students/2      
-
+The DESTROY method will delete student row with the primary key "2". Fun fact, if the same student were to be recreated, it would be assigned a new primary key and the primary key "2" would be permanently unusable.
 
 
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
```

Comment:
User stories can be tricky, especially for developers because our thoughts are on creating the code that will accomplish the task. User stories are more about the user experience with the DOM. What the user can see, click, type...how they get to interact with the web page.